server/push_test.go

```diff
@@ -27,19 +34,154 @@ func requestSetup(s *LivepeerServer) (http.Handler, *strings.Reader, *httptest.R
 	return handler, reader, writer
 }
 
-func TestNoErrors(t *testing.T) {
```

Comment:
Don't we also want to test non-multipart returns? Isn't that test case being removed here?

Reply:
This case is handled in `TestMultipartReturn` test under `// Body should be empty if no Accept header specified` comment. And `TestNoErrors`, as it is, was not working anymore - so I've just removed it.

Reply:
> And TestNoErrors, as it is, was not working anymore - so I've just removed it.

What made it stop working? Surely existing behavior shouldn't be changed?

Reply:
This addition:
```golang
	if tr == nil {
		http.Error(w, "No sessions available", http.StatusServiceUnavailable)
		return
	}
```
made it to fail. Previously there was no indication for caller that segment actually was not processed. Don't think we should preserve old behaviour.